### ISSUE TITLE
Let It Snow! Let It Snow! Let It Snow!

### ISSUE THREAD

**Issue by brettpenzer123** [OpenRCT2/OpenRCT2#3576]
Just making some final adjustments.

![s](https://cloud.githubusercontent.com/assets/10356800/15218871/a794970a-1859-11e6-86ab-a8075fca38df.png)

**Comment by Gymnasiast** [comment]
Snow with a temperature of 17°C?

**Comment by brettpenzer123** [comment]
Obviously I haven't implemented the temperature change but I'll nearly done :dancer:

**Comment by brettpenzer123** [comment]
The sprite was already implemented.

**Comment by spacek531** [comment]
Could the snowflakes on the screen be enlarged a little? It's not very easy to tell what they are.

**Comment by chrismilne300** [comment]
It would be amazing if the ground changed like it does in Locomotion:
![screen shot 05-12-16 at 05 47 pm](https://cloud.githubusercontent.com/assets/1381723/15222924/c4c67086-1869-11e6-9aea-48b183129486.PNG)
![screen shot 05-12-16 at 05 48 pm](https://cloud.githubusercontent.com/assets/1381723/15222925/c4ca2c3a-1869-11e6-9b69-c86e6bd76251.PNG)

**Comment by Superjustinbros** [comment]
Perhaps the snow could be made into some tiny star-shaped sprites and instead of the screen getting darker like in the rain, it gets brighter and more pale.

**Comment by IntelOrca** [comment]
Shouldn't this be a pull request rather than an issue?
We have a `work in progress` label for pull requests.

**Comment by WhenPenguinsAttack** [comment]
I agree that the snow should be bigger so we can see it easier, and also agree with the ground changes so that we can see built up snow.

**Comment by brettpenzer123** [comment]
I've made the snow drops bigger then the ones in previous image. I'm also adding in some weather effects that can be toggled optional like rain that makes flowers bloom again and snow that can form ice on grass. If we had more terrain options I'd gladly add more surface type snow.

**Comment by Ziscor** [comment]
Here's a thought: the snow should fall slower than rain does (unless that's exactly what you've done). If the snow were to fall at the rate at which it does now it would be a damn blizzard :snowflake: 
It should be gliding/ drifting across the screen.

**Comment by Superjustinbros** [comment]
@Ziscor I like that idea.

**Comment by brettpenzer123** [comment]
I've decided to remove bigger snowflakes as it was getting in the way. The snow falls at a 125 degree angle which looks nicer as the first time I did it. It was zigzagging slowly coming down.

![sds3](https://cloud.githubusercontent.com/assets/10356800/15232985/d5a77e44-189c-11e6-82d1-807f662318c0.gif)

**Comment by brettpenzer123** [comment]
Currently it runs within `drawing.c function gfx_draw_rain` and `rain.c function draw_light_rain` which isn't very optimal but I'm going to make its own functions and calls in time.

**Comment by IntelOrca** [comment]
How can you change the names of the climates to seasons? The climate is fixed to the scenario, seasons would happen every game year, and parks aren't usually open in the winter which is why the months run from March to October (assuming northern hemisphere).

**Comment by brettpenzer123** [comment]
The game has strings available for Jan to Dec it can run a climate every three months in game. Parks wouldn't be open during the weather `heavy_rain` but we don't close rides nor do we close the park. So it doesn't matter if a park wouldn't open in winter nor snow.

It's a simulation game not a simulator.

**Comment by IntelOrca** [comment]
In the game a climate is a fixed scheme of weather and temperature patterns. It can not change throughout a scenario. You have:
```
CLIMATE_COOL_AND_WET,
CLIMATE_WARM,
CLIMATE_HOT_AND_DRY,
CLIMATE_COLD
```

You said you changed those to seasons... but they aren't. Each one of those climates defines a pattern of temperatures throughout the year thus, simulating the seasons. The idea is that you make a desert map to `CLIMATE_HOT_AND_DRY` and a ice map to `CLIMATE_COLD`. You wouldn't expect snow ever on a `CLIMATE_HOT_AND_DRY` map, but to get it frequently on a `CLIMATE_COLD` map.

**Comment by brettpenzer123** [comment]
I understand but if change one I must change all.

**Comment by brettpenzer123** [comment]
I was looking at pull requesting the snow weather but I have to sort out what isn't needed for the development version. Most of my code is conflicts :)

**Comment by brettpenzer123** [comment]
Most of my code isn't pull requested anyways and will more then likely stay in my development.

**Comment by IntelOrca** [comment]
> I understand but if change one I must change all.
I don't know what you mean.

> I was looking at pull requesting the snow weather but I have to sort out what isn't needed for the development version. Most of my code is conflicts :)
I can't see any commits on your repository for this which suggests you haven't pushed your local changes to GitHub. It means no one can see what you have done or suggest anything. Pull requests are a good way of tracking the changes and testing if its mergable.

> Most of my code isn't pull requested anyways and will more then likely stay in my development.
This doesn't make any sense.

**Comment by brettpenzer123** [comment]
My whole post above was to state that my code is way out of sync with the rest of the project.

**Comment by IntelOrca** [comment]
@brettpenzer123 read this:

https://guides.github.com/introduction/flow/

**Comment by brettpenzer123** [comment]
Thanks for the information but I decided not to push changes to my fork. Therefore my code is way out which will take me longer to convert it from one source to the other. When I'm done moving the code over then I will start fresh.

When I'm trying to implement a new feature I don't want any of the global variables or other code not working properly. This is why I prefer to just work offline.

**Comment by IntelOrca** [comment]
@brettpenzer123 that's not how this works and I think you are misunderstanding this. Your fork is your own version of this repository. You can do what you like with it and anything anyone else does will not interfere with it. You should never ever need to start fresh, that defeats the whole principle of git and source / version control.

Have a look at https://guides.github.com/activities/forking/

**Comment by brettpenzer123** [comment]
I know that isn't how it works. This is just my preference.

**Comment by janisozaur** [comment]
What was your point in opening an issue and showcasing something you apparently are not willing to contribute?

**Comment by IntelOrca** [comment]
@brettpenzer123 That's unfortunate to hear as that goes against ours and GitHub's contribution etiquette. I really hope you reconsider.

**Comment by brettpenzer123** [comment]
@janisozaur I didn't say I'd never add it. I just decided to not use the standard procedures.

@IntelOrca That's fine. I did say once I get it into the source you are currently merging I'll happily get on wagon. Now I'm rewriting all the code because most of the variables that I used are changed or removed. If anything I should be the one complaining because its going to take longer to pull request.

My method equals time due to the massive amount of change. I understand the proper method...
